app/models/post.rb

```diff
@@ -1,2 +1,15 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+
+  has_many :comments, foreign_key: :post_id
+
+  has_many :likes, foreign_key: :post_id
+
+  def update_post_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] Optional:  It would be nice to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private.

Example:
```
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```